libnmstate/nm/translator.py

```diff
@@ -59,6 +59,35 @@ def get_bond_options(bond_conf):
 class Nm2Api(object):
     _iface_types_map = None
 
+    @staticmethod
+    def get_common_device_info(devinfo):
+        type_name = devinfo['type_name']
+        if type_name != 'ethernet':
+            type_name = Nm2Api.get_iface_type(type_name)
+        return {
+            'name': devinfo['name'],
+            'type': type_name,
+            'state': Nm2Api.get_iface_admin_state(devinfo['state']),
+        }
+
+    @staticmethod
+    def get_bond_info(bondinfo):
+        bond_options = bondinfo.get('options')
+        if not bond_options:
+            return {}
+        bond_slaves = bondinfo['slaves']
+
+        bond_mode = bond_options['mode']
+        del bond_options['mode']
```

Comment:
doing this you change original object passed as argument, it would be safer to create a copy.